qiskit/algorithms/gradients/reverse_gradient/reverse_qgt.py

```diff
@@ -0,0 +1,252 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""QGT with the classically efficient reverse mode."""
+
+from __future__ import annotations
+from collections.abc import Sequence
+import logging
+
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit, Parameter
+from qiskit.quantum_info import Statevector
+from qiskit.providers import Options
+from qiskit.primitives import Estimator
+
+from ..base_qgt import BaseQGT
+from ..qgt_result import QGTResult
+from ..utils import DerivativeType
+
+from .split_circuits import split
+from .bind import bind
+from .derive_circuit import derive_circuit
+
+logger = logging.getLogger(__name__)
+
+
+class ReverseQGT(BaseQGT):
+    """QGT calculation with the classically efficient reverse mode.
+
+    .. note::
+
+        This QGT implementation is based on statevector manipulations and scales exponentially
+        with the number of qubits. However, for small system sizes it can be very fast
+        compared to circuit-based gradients.
+
+    This class implements the calculation of the QGT as described in [1].
+    By keeping track of three statevectors and iteratively sweeping through each parameterized
+    gate, this method scales only quadratically with the number of parameters.
+
+    **References:**
+
+        [1]: Jones, T. "Efficient classical calculation of the Quantum Natural Gradient" (2020).
+             `arXiv:2011.02991 <https://arxiv.org/abs/2011.02991>`_.
+
+    """
+
+    SUPPORTED_GATES = ["rx", "ry", "rz", "cp", "crx", "cry", "crz"]
+
+    def __init__(
+        self, phase_fix: bool = True, derivative_type: DerivativeType = DerivativeType.COMPLEX
+    ):
+        """
+        Args:
+            phase_fix: Whether or not to include the phase fix.
+            derivative_type: Determines whether the complex QGT or only the real or imaginary
+                parts are calculated.
+        """
+        dummy_estimator = Estimator()  # this method does not need an estimator
```

Comment:
This is just a comment. The only use in the base class seems to be for merging options - something again it seems you needed to override below in order to produce an empty entity rather than having the superclass including whatever options are given back by the dummy Estimator. Maybe the base class should have been without an Estimator to facilitate extension - but maybe this classical extension is a exception and things are fine as they are.

Reply:
Yeah I thought about that too... other than this classical algorithm only methods come to mind that do need an estimator. However the base class should maybe only include implementation-independent specifications on the QGT like the derivative type or the phase fix, and nothing else.

Reply:
From what I could see the base class only references the estimator its passed to support the options. Aside from merging options it is passed with that of the estimator it was not used all all.

Reply:
On the other hand, it would be good to be consistent and the `BaseEstimatorGradient` with the `Estimator` argument in the base class is already released so we would have to deprecate it there before removing it.

Reply:
I think it's ok for now as things - so I'll resolve this and approve the PR